src/metabase/formatter.clj

```diff
@@ -265,22 +266,39 @@
              (format "%.8f° %s" (Math/abs v) dir)
              v)))))
 
+(defn dictionary-formatter
+  "Format dictionaries as json.
+  The value if a dictionary is Clojure edn on the backend, but displays as JSON in
+  When exporting, the map must be encoded as json so that exports match the app's output."
+  [value]
+  (if (string? value)
+    value
+    (json/encode value)))
+
 (mu/defn create-formatter
   "Create a formatter for a column based on its timezone, column metadata, and visualization-settings"
-  [timezone-id :- [:maybe :string] col visualization-settings]
-  (cond
-    ;; for numbers, return a format function that has already computed the differences.
-    ;; todo: do the same for temporal strings
-    (types/temporal-field? col)
-    #(datetime/format-temporal-str timezone-id % col visualization-settings)
+  ([timezone-id :- [:maybe :string] col visualization-settings]
+   (create-formatter timezone-id col visualization-settings true))
+  ([timezone-id :- [:maybe :string] col visualization-settings apply-formatting?]
+   (cond
+     ;; for numbers, return a format function that has already computed the differences.
+     ;; todo: do the same for temporal strings
+     (and apply-formatting? (types/temporal-field? col))
+     #(datetime/format-temporal-str timezone-id % col visualization-settings)
+
+     (and apply-formatting? (isa? (:semantic_type col) :type/Coordinate))
+     (partial format-geographic-coordinates (:semantic_type col))
 
-    (isa? (:semantic_type col) :type/Coordinate)
-    (partial format-geographic-coordinates (:semantic_type col))
+     ;; todo integer columns with a unit
+     (and apply-formatting?
+          (or (isa? (:effective_type col) :type/Number)
+              (isa? (:base_type col) :type/Number)))
+     (number-formatter col visualization-settings)
 
-    ;; todo integer columns with a unit
-    (or (isa? (:effective_type col) :type/Number)
-        (isa? (:base_type col) :type/Number))
-    (number-formatter col visualization-settings)
+     (or (isa? (:semantic_type col) :type/SerializedJSON)
+         (isa? (:base_type col) :type/Dictionary)
+         (isa? (:effective_type col) :type/Dictionary))
```

Comment:
I think this should actually be something like 

```clj
(isa? (some-fn :effective_type :base_type) :type/Dictionary)
```

rather than `(or <isa? base-type...> <isa? effective-type...>)`... 

The reason for this is that effective type is how we actually treat the column whereas base type is the actual underlying column type, for example in the future it might be possible that we have a column of base type `:type/Dictionary` that undergoes a transformation to pull out a specific key giving it an effective type of `:type/Number` which means that we definitely shouldn't be applying map rules to that value.

This is not true now but it is the way this stuff is supposed to work so it would be nice that if we add that in the future it doesn't break this code